src/sentry/api/serializers/models/group_stream.py

```diff
@@ -338,6 +338,11 @@ def query_tsdb(
         perf_issue_ids = [
             group.id for group in groups if GroupCategory.PERFORMANCE == group.issue_category
         ]
+        generic_issue_ids = [
+            group.id
+            for group in groups
+            if group.issue_category not in (GroupCategory.ERROR, GroupCategory.PERFORMANCE)
+        ]
```

Comment:
Nit: Could be worth maybe factoring this a bit more:

```
for group_filter, tsdb_model in [
    (lambda group: GroupCategory.ERROR == group.issue_category, snuba_tsdb.models.group),
    (lambda group: GroupCategory.PERFORMANCE == group.issue_category, snuba_tsdb.models.group_performance),
    (lambda group: group.issue_category not in (GroupCategory.ERROR, GroupCategory.PERFORMANCE), snuba_tsdb.models.group_generic),
]:
    results.update(get_range(model=tsdb_model, keys=group_filter(groups)))
```

Not totally sure if better

Reply:
Yeah, this bit of code can definitely be more concise. Took a whack at cleaning up the partitioning the groups into the lists.